server/test/test-utils.ts

```
@@ -41,135 +34,58 @@ export const db = {
+      queue: (item: JobItem) => runJobs && _handler(item),
+      resume: jest.fn(),
+      empty: jest.fn(),
+      setConcurrency: jest.fn(),
+      getQueueStatus: jest.fn(),
+      getJobCounts: jest.fn(),
+      pause: jest.fn(),
+    } as IJobRepository);
+
+  if (log) {
+    moduleBuilder.setLogger(new Logger());
+  }
+
+  const moduleFixture: TestingModule = await moduleBuilder.compile();
+
+  const app = await moduleFixture.createNestApplication().init();
+  const appService = app.get(AppService);
+  await appService.init();
+
+  return app;
 }
 
-export const api = {
-  adminSignUp: async (server: any) => {
-    const { status, body } = await request(server).post('/auth/admin-sign-up').send(adminSignupStub);
-
-    expect(status).toBe(201);
-    expect(body).toEqual(signupResponseStub);
-
-    return body as AdminSignupResponseDto;
-  },
-  adminLogin: async (server: any) => {
-    const { status, body } = await request(server).post('/auth/login').send(loginStub.admin);
-
-    expect(body).toEqual(loginResponseStub.admin.response);
-    expect(body).toMatchObject({ accessToken: expect.any(String) });
-    expect(status).toBe(201);
-
-    return body as LoginResponseDto;
-  },
-  userCreate: async (server: any, accessToken: string, user: Partial<UserEntity>) => {
-    const { status, body } = await request(server)
-      .post('/user')
-      .set('Authorization', `Bearer ${accessToken}`)
-      .send(user);
+const directoryExists = async (dirPath: string) =>
+  await fs.promises
+    .access(dirPath)
+    .then(() => true)
+    .catch(() => false);
 
-    expect(status).toBe(201);
+export async function ensureTestAssets(): Promise<void> {
+  if (!(await directoryExists(`${TEST_ASSET_PATH}/albums`))) {
+    const errorString = `Test assets not found. Please checkout https://github.com/immich-app/test-assets into ${TEST_ASSET_PATH} before testing`;
+    throw new Error(errorString);
```

I think you can throw the error in here directly, not create a separate string first.

---

Strangely enough, I wasn't able to get that to work, it just did not do the variable substitution. Will try again